src/objectcontroller.cpp

```diff
@@ -1478,6 +1484,303 @@ void DateObject::getDate(int *day, int *month, int *year)
         *year = 1900;
 }
 
+DateTimeObjectValue::DateTimeObjectValue(const std::string& value) : 
+  day_m(-1), month_m(-1), year_m(-1), hour_m(-1), min_m(-1), sec_m(-1)
+{
+    if (value == "now")
+        return;
+    std::istringstream val(value);
+    char s1, s2, s3, s4, s5;
+    val >> year_m >> s1 >> month_m >> s2 >> day_m >> s3 >> hour_m >> s4 >> min_m >> s5 >> sec_m;
+    wday_m = 0;
+    year_m -= 1900;
+    if ( val.fail() ||
+         val.peek() != std::char_traits<char>::eof() || // workaround for wrong val.eof() flag in uClibc++
+         s1 != '-' || s2 != '-' || s3 != 'T' || s4 != ':' || s5 != ':' ||
+         year_m < 0 || year_m > 255 || month_m < 1 || month_m > 12 || day_m < 1 || day_m > 31 ||
+	 hour_m < 0 || hour_m > 23 || min_m < 0 || min_m > 59 || sec_m < 0 || sec_m > 59 )
+    {
+        std::stringstream msg;
+        msg << "DateTimeObjectValue: Bad value: '" << value << "'" << std::endl;
+        throw ticpp::Exception(msg.str());
+    }
+}
+
+std::string DateTimeObjectValue::toString() const
+{
+    if (day_m == -1)
+        return "now";
+    std::ostringstream out;
+    out << year_m+1900 << "-" << month_m << "-" << day_m << 'T'
+      << std::setfill('0') << std::setw(2) 
+      << hour_m << ":" << std::setfill('0') << std::setw(2) 
+      << min_m << ":" << std::setfill('0') << std::setw(2)
+      << sec_m;
+    return out.str();
+}
+
+double DateTimeObjectValue::toNumber()
+{
+    if (day_m == -1)
+        return -1.0;
+    return (year_m * 400 + month_m * 31 + day_m) * 86400 +
+            hour_m * 3600 + min_m * 60 + sec_m;
+}
+
+bool DateTimeObjectValue::equals(ObjectValue* value)
+{
+    int day, month, year, wday, hour, min, sec;
+    assert(value);
+    DateTimeObjectValue* val = dynamic_cast<DateTimeObjectValue*>(value);
+    if (val == 0)
+    {
+        logger_m.errorStream()  << "DateTimeObjectValue: ERROR, equals() received invalid class object (typeid=" << typeid(*value).name() << ")" << endlog;
+        return false;
+    }
+    // logger_m.debugStream() << "DateTimeObjectValue (id=" << getID() << "): Compare value_m='" << value_m << "' to value='" << val->value_m << "'" << endlog;
+    val->getDateTimeValue(&day, &month, &year, &wday, &hour, &min, &sec);
+    return (day_m == day) && (month_m == month) && (year_m == year) && (sec_m == sec) && (min_m == min) && (hour_m == hour);
```

Comment:
For the sake of simplicity and to help reduce the amount of copy & paste in this codebase:
```cpp
bool DateTimeObjectValue::equals(ObjectValue* value)
{
    return compare(value) == 0;
}
```

You can argue that this remarks is true for all other objects and I would totally agree. There is certainly no point in marking `equals` as virtual. I may clean that up in the future. But let's at least not repeat the past errors for this new class ;)